src/helpers/presentation.rs

```diff
@@ -0,0 +1,323 @@
+    #[error("internal error occurred; {0}")]
+    InternalError(String),
+    #[error(r#"JSON serialization of {0} failed due to "{1}""#)]
+    JsonSerialization(String, String),
+    #[error(r#"JSON deserialization of {0} failed due to "{1}" on: {2}"#)]
+    JsonDeserialization(String, String, String),
+    #[error(r#"schema with DID "{0}" does not seem to be a valid schema: {1}"#)]
+    SchemaInvalid(String, String),
+    #[error(r#"schema with DID "{0}" could not be found"#)]
+    SchemaNotFound(String),
+}
+
+impl PresentationError {
+    fn to_serialization_error(
+        parsed_object: &str,
+    ) -> impl Fn(serde_json::Error) -> PresentationError + '_ {
+        move |err| PresentationError::JsonSerialization(parsed_object.to_string(), err.to_string())
+    }
+
+    fn to_deserialization_error<'a>(
+        to_parse_name: &'a str,
+        to_parse_value: &'a str,
+    ) -> impl Fn(serde_json::Error) -> PresentationError + 'a {
+        move |err| {
+            PresentationError::JsonDeserialization(
+                to_parse_name.to_string(),
+                err.to_string(),
+                to_parse_value.to_string(),
+            )
+        }
+    }
+}
+
+// Master secret is always incorporated, without being mentioned in the credential schema
+const ADDITIONAL_HIDDEN_MESSAGES_COUNT: usize = 1;
+const NQUAD_REGEX: &str = r"^_:c14n0 <http://schema.org/([^>]+?)>";
+const TYPE_OPTIONS: &str = r#"{ "type": "bbs" }"#;
+
+pub struct Presentation<'a> {
+    vade_evan: &'a mut VadeEvan,
+}
+
+impl<'a> Presentation<'a> {
+    pub fn new(vade_evan: &'a mut VadeEvan) -> Result<Self, PresentationError> {
+        Ok(Self { vade_evan })
+    }
+
+    pub async fn create_proof_request(
+        &mut self,
+        schema_did: &str,
+        revealed_attributes: Option<&str>,
+    ) -> Result<String, PresentationError> {
+        let revealed_attributes_parsed: Option<Vec<String>> = revealed_attributes
+            .map(|ras| {
+                serde_json::from_str(ras).map_err(PresentationError::to_deserialization_error(
+                    "revealed attributes",
+                    ras,
+                ))
+            })
+            .transpose()?;
+        let proof_request_payload = RequestProofPayload {
+            verifier_did: None,
+            schemas: vec![schema_did.to_string()],
+            reveal_attributes: self
+                .get_reveal_attributes_map(schema_did, revealed_attributes_parsed)
+                .await?,
+        };
+
+        let proof_request_json = serde_json::to_string(&proof_request_payload).map_err(
+            PresentationError::to_serialization_error("RequestProofPayload"),
+        )?;
+
+        self.vade_evan
+            .vc_zkp_request_proof(EVAN_METHOD, TYPE_OPTIONS, &proof_request_json)
+            .await
+            .map_err(|err| PresentationError::VadeEvanError(err.to_string()))
+    }
+
+    async fn get_did_document<T>(&mut self, did: &str) -> Result<T, PresentationError>
```

Comment:
We have the same function in the `helper/credential.rs`. It seems like you have a bit better error handling here. Would it make sense to move yours into the `shared.rs` and use it in both situations?